fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryScoreForRankPlan.java

```diff
@@ -139,24 +150,32 @@ public String toString() {
         return getChild() + " WHERE " + ranks.stream().map(Object::toString).collect(Collectors.joining(", "));
     }
 
+    @Nonnull
+    @Override
+    @API(API.Status.EXPERIMENTAL)
+    public Set<CorrelationIdentifier> getCorrelatedToWithoutChildren() {
+        return ImmutableSet.of();
+    }
+
+    @Nonnull
     @Override
     @API(API.Status.EXPERIMENTAL)
-    public boolean equalsWithoutChildren(@Nonnull RelationalExpression otherExpression) {
+    public RecordQueryScoreForRankPlan rebaseWithRebasedQuantifiers(final Map<CorrelationIdentifier, CorrelationIdentifier> translationMap, final List<Quantifier> rebasedQuantifiers) {
+        return new RecordQueryScoreForRankPlan((Quantifier.Physical)Iterables.getOnlyElement(rebasedQuantifiers),
+                getRanks());
+    }
+
+    @Override
+    @API(API.Status.EXPERIMENTAL)
+    public boolean equalsWithoutChildren(@Nonnull RelationalExpression otherExpression, final BiMap<CorrelationIdentifier, CorrelationIdentifier> equivalencesMap) {
         return otherExpression instanceof RecordQueryScoreForRankPlan &&
                ranks.equals(((RecordQueryScoreForRankPlan) otherExpression).ranks);
     }
 
+    @SuppressWarnings("EqualsWhichDoesntCheckParameterClass")
     @Override
-    public boolean equals(Object o) {
-        if (this == o) {
-            return true;
-        }
-        if (o == null || getClass() != o.getClass()) {
-            return false;
-        }
-        RecordQueryScoreForRankPlan that = (RecordQueryScoreForRankPlan) o;
-        return Objects.equals(getChild(), that.getChild()) &&
-                Objects.equals(ranks, that.ranks);
+    public boolean equals(final Object other) {
+        return resultEquals(other, ImmutableBiMap.of());
```

Comment:
This has changed in general since you reviewed this. `resultEquals()` is only implemented in `RelationalExpression` which is much nicer. That generic implementation then does the necessary things and defers to `equalsWithoutChildren()` after establishing result equality of the children.